src/i18n/index.ts

```diff
@@ -0,0 +1,22 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import en from './en.json';
+import es from './es.json';
+
```

Comment:
Please extend the default type provided by i18n to include our translations. This will help in getting suggestions using Intellisense when typing translation keys inside `t()`.

You could either include the following segment of code in this file or checkout the alternate approaches [here](https://www.i18next.com/overview/typescript#create-a-declaration-file).

```
declare module 'i18next' {
    // Extend CustomTypeOptions
    interface CustomTypeOptions {
        defaultNS: 'en';
        resources: {
            en: typeof en;
            es: typeof es;
        };
    }
}
```

Finallly, you could see the difference by start typing inside the `t()` function. For example, in the header file if you start typing, "home" should come as suggestion and it will show an error if you pass any non key as its parameter. I hope this additional typescript check will make further development easier and make it less error prone.

Reply:
Resolved in 6494fa981a80486a4f5325c40214a1ec40a34a2e

Reply:
I have also added namespaces for better modularity.